build.gradle

```diff
@@ -108,18 +134,39 @@ task executeCmake(type: Exec) {
 
     workingDir "${buildDir}/cmake"
     def prebuiltJar = null
-    if (project.hasProperty('stagingUrl')) {
+    def projectName = isFips ? project.name + "-FIPS" : project.name
+    def prebuiltJarFileName = "${projectName}-${version}-${osdetector.classifier}.jar"
+    if (System.properties['stagingProperties'] != null) {
+        def stagingProperties = new Properties()
+        if (isFips) {
+            stagingProperties.load(
+                new java.io.FileInputStream("${System.env.CODEBUILD_SRC_DIR_Stage_FIPS}/fake_maven/staging.properties"));
+        } else {
+            stagingProperties.load(
+                new java.io.FileInputStream("${System.env.CODEBUILD_SRC_DIR_Stage}/fake_maven/staging.properties"));
```

Comment:
why are staging properties different between FIPS/non-FIPS?

Reply:
Because FIPS and non-FIPS have different repos and staging Ids. It seems impractical to combine them.